lib/variable/operations_common.h

```diff
@@ -35,8 +36,17 @@ template <class T>
 Variable make_bins_impl(Variable indices, const Dim dim, T &&buffer);
 
 template <class T, class Op> auto reduce_all_dims(const T &obj, const Op &op) {
-  if (obj.dims().empty())
-    return copy(obj);
+  if (obj.dims().empty()) {
+    bool bins = false;
+    if constexpr (std::is_same_v<T, Variable>)
```

Comment:
Doesn't `is_bins` take care of this check?

Reply:
`is_bins` does not support `DataArray`.

Reply:
It does: https://github.com/scipp/scipp/blob/925c25102200e7f4ff79ebff01e7c32f7e0bec5b/lib/dataset/include/scipp/dataset/bins.h#L36

But this declaration is not available in `variable`. So I guess the current solution is ok...

Reply:
Hmm, maybe just a missing include that caused a compile failure...

Reply:
See #2686